constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/ConstraintSetParser.java

```diff
@@ -0,0 +1,1201 @@
+                    return margins.get(stringValue).floatValue();
+                }
+            } else if (elementName instanceof CLNumber) {
+                return ((CLNumber) elementName).getFloat();
+            }
+            return 0f;
+        }
+
+        ArrayList<String> getList(String elementName) {
+            if (arrayIds.containsKey(elementName)) {
+                return arrayIds.get(elementName);
+            }
+            return null;
+        }
+
+        void put(String elementName, ArrayList<String> elements) {
+            arrayIds.put(elementName, elements);
+        }
+
+    }
+
+    interface GeneratedValue {
+        float value();
+    }
+
+    static class Generator implements GeneratedValue {
+        float mStart = 0;
+        float mIncrementBy = 0;
+        float mCurrent = 0;
+        boolean mStop = false;
+
+        Generator(float start, float incrementBy) {
+            mStart = start;
+            mIncrementBy = incrementBy;
+            mCurrent = start;
+        }
+
+        @Override
+        public float value() {
+            if (!mStop) {
+                mCurrent += mIncrementBy;
+            }
+            return mCurrent;
+        }
+    }
+
+    static class FiniteGenerator implements GeneratedValue {
+        float mFrom = 0;
+        float mTo = 0;
+        float mStep = 0;
+        boolean mStop = false;
+        String mPrefix;
+        String mPostfix;
+        float mCurrent = 0;
+        float mInitial;
+        float mMax;
+
+        FiniteGenerator(float from,
+                        float to,
+                        float step,
+                        String prefix,
+                        String postfix) {
+            mFrom = from;
+            mTo = to;
+            mStep = step;
+            mPrefix = prefix;
+            mPostfix = postfix;
+            mMax = to;
+            mInitial = from;
+        }
+
+        @Override
+        public float value() {
+            if (mCurrent >= mMax) {
+                mStop = true;
+            }
+            if (!mStop) {
+                mCurrent += mStep;
+            }
+            return mCurrent;
+        }
+
+        public ArrayList<String> array() {
+            ArrayList<String> array = new ArrayList<>();
+            int value = (int) mInitial;
+            int maxInt = (int) mMax;
+            for (int i = value; i <= maxInt; i++) {
+                array.add(mPrefix + value + mPostfix);
+                value += mStep;
+            }
+            return array;
+
+        }
+
+    }
+
+    static class OverrideValue implements GeneratedValue {
+        float mValue;
+
+        OverrideValue(float value) {
+            mValue = value;
+        }
+
+        @Override
+        public float value() {
+            return mValue;
+        }
+    }
+//==================== end store variables =========================
+//==================== MotionScene =========================
+
+    public enum MotionLayoutDebugFlags {
```

Comment:
nit: MotionLayoutDebugFlag

singular instead of plural

Reply:
Removed